modules/internal/String.chpl

```diff
@@ -675,15 +703,21 @@ module String {
     proc type chpl__deserialize(data) {
       if data.locale_id != chpl_nodeID {
         if data.len <= CHPL_SHORT_STRING_SIZE {
-          return createStringWithNewBuffer(
+          try! {
```

Comment:
I don't think this one needs to validate, because the string was created on the other locale and should have validated by then. You don't necessarily need to fix in this PR but I think a comment about it would be helpful.

Reply:
You're right. I think `chpl__deserialize` and `localize` (as you commented below) can use `chpl_createString`. I'll make that change here as it is less controversial :)